pkg/registry/apis/provisioning/github_worker.go

```diff
@@ -0,0 +1,72 @@
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/auth"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/jobs"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/repository"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/resources"
+)
+
+var (
+	_ jobs.Worker = (*GithubWorker)(nil)
+)
+
+type GithubWorker struct {
```

Comment:
This file is currently in the root directory so we can avoid import cycles.  I think getting the key interfaces into an importable package could help with that

Reply:
This is the reason why I extracted out the webhook handling out the repository into a `webhook` package. I still think that parsing and processing should be extracted out the repository package.

Let's add a FIXME comment for now to discuss on that structure on a follow-up.